Documentation/pre-built-detection.md

```diff
@@ -0,0 +1,147 @@
+In lament terms, this means that packages from `nuget.org`, Microsoft builds or other unrelated source-builds cannot be used for source-building a given repository.
+
+Several different methods are provided to repositories to be able to handle pre-builts for their individual builds while being source-buildable.
+These methods are expanded on in the [Eliminating pre-builts](#eliminating-pre-builts) section of the guide.
+
+# Enabling mandatory pre-built detection
+
+## Motivation
+
+WIP
+
+## Verifying that the repository is source-buildable (?)
+
+WIP
+
+## Identifying used pre-builts
+
+By default, when source-build logic was first introduced to the repository, all pre-builts were allowed.
+This was accomplished by instructing pre-built detection logic to not fail the build for any found pre-builts via a dedicated glob rule.
+The rule itself can be found in the `./eng/SourceBuildPrebuildBaseline.xml` file.
+
+Nevertheless, pre-built detection compiles a report of pre-built usage regardless of any rules set.
+To receive the afore-mentioned report, the repository must be source-build first.
+This can be done via the following command:
+
+```
+./build.sh --sb --bl
+```
+
+Once the source-build succeddes, several files related to pre-built detection can be found in the `./artifacts/source-build/self/prebuild-report` directory.
+For now, the one that is of particular interest is `./prebuild-usage.xml`.
+It contains a list of pre-builts discovered during the build process together with a path to the project that requires the pre-built in question. 
+
+If the list is empty - the build does not utilize any pre-builts. 
+However, in most cases it will contain several of them and will require additional actions before mandatory pre-built detection can be enabled.
+
+# Eliminating pre-builts
+
+To better understand how a pre-built can be tackled, first its type should be identified. 
+
+In general, project dependencies can be split into the following categorizes:
+  - direct - dependency of the repository itself, i.e. declared in its `./eng/Version.Details.xml` file
+  - indirect (transitive) - dependencies of a direct dependency that are required by the later
+  - dangling - dependencies that are not part of the `restore` during repository build but are installed as part of the build by, for example, some tooling
+  - external - dependency on a component outside of Microsoft / .NET
+
+Knowing what exactly is the pre-built in question helps greatly in eliminating it from the report based on the following flow:
+
+  1. [Direct dependency from a Microsoft / .NET repository](#direct-dependency-from-a-microsoft--net-repository)
+  2. [Dependencies witout any code](#dependencies-without-any-code)
+  3. [External dependencies](#external-dependencies)
+
+## Direct dependency from a Microsoft / .NET repository
+
+When a repository depends on package from another Microsoft / .NET repository, existence of a source-build for the latter should be checked.
```

Comment:
We should be sharing the [existing dependency documentation](https://github.com/dotnet/source-build/tree/main/Documentation/sourcebuild-in-repos) as much as possible.  I see a fair amount of overlap here.  If the existing doc needs edits to work here then they should be made.

Reply:
Agreed, simplified and rephrased the section a bit to point to the doc. I think it structured well enough to be used for this use-case, maybe just some things like https://github.com/dotnet/source-build/pull/3083#discussion_r1029759376 can be done there + some sections can be expanded and given examples to illustrate the work required.